packages/flutter_tools/lib/src/linux/linux_doctor.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/io.dart';
+import '../base/process_manager.dart';
+import '../doctor.dart';
+
+/// A validator that checks for GCC and the correct gtk version.
+class LinuxDoctorValidator extends DoctorValidator {
+  const LinuxDoctorValidator() : super('Linux toolchain - develop for Linux desktop applications');
+
+  @override
+  Future<ValidationResult> validate() async {
+    ValidationType validationType = ValidationType.installed;
+    final List<ValidationMessage> messages = <ValidationMessage>[];
+    /// Check for some version of GCC.
+    final ProcessResult gccResult = await processManager.run(const <String>[
+      'g++',
```

Comment:
The choice of GCC vs clang is specific to the FDE example (and might be something we want to change before templatizing); it doesn't seem like we should have `doctor` check for things that aren't required by `flutter_tool`.

How about checking for `make`, since that's what the tooling actually expects at this point?

Reply:
The doctor could check make and then for both clang/gcc as a hint? Longer term, I would like to break out each doctor validation check to make it platform/template specific (see go/flutter-tool-extensions).

Reply:
I'm not familiar yet with the gradations of checks that the doctor support, but an error for not having make and a warning for not having gcc or clang definitely seems reasonable as a starting point. (Substituting whatever the right terminology is here for 'error' and 'warning')